### DISCUSSION TITLE
请问vue组件引入文件，在打包以后如何支持替换？

### DISCUSSION THREAD

**Discussion by TeddyQ** [alex8088/electron-vite#307] [Q&A]
关于资源处理，asset 和 asarUnpack 只能用于主进程吗？

我在vue组件中导入svg文件：
import svgUrl from '../../assets/test.svg?asset&asarUnpack'

打包以后查看DOM，文件加载的是app.asar里的内容：
<embed src="file:///D:/electron/ros/dist/win-unpacked/resources/app.asar/out/renderer/assets/test-7d0548dd.svg?asset&amp;asarUnpack">

请问怎样可以不加载app.asar里的文件，并且文件支持可替换呢？

**Comment by TeddyQ** [comment]
打包以后查看DOM，文件加载的是app.asar里的内容：
<标签embed src="file:///D:/electron/ros/dist/win-unpacked/resources/app.asar/out/renderer/assets/test-7d0548dd.svg?asset&amp;asarUnpack" >

**Comment by alex8088** [comment]
资源处理
本章节中的资源处理功能仅适用于主进程。关于渲染进程的静态资源处理，请参阅 Vite 的 [静态资源处理](https://cn.vitejs.dev/guide/assets.html)。

**Comment by alex8088** [comment]
渲染进程资源要动态处理，不想打包到asar中可以通过自定义协议处理 参考 https://github.com/alex8088/electron-vite/issues/36
